### PR TITLE
Ledger pagination and GST invoice download implemented

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts [hunk 1]
```diff
@@ -234,4 +246,92 @@ export class ViewTransactionsComponent implements OnInit, OnChanges, OnDestroy {
...
+    * @param invoiceUniqueName
+    * @memberof ViewTransactionsComponent
+    */
+    public downloadFile() {
+        let blob = this.base64ToBlob(this.base64Data, 'application/pdf', 512);
+        return saveAs(blob, `Invoice-${this.selectedInvoice.account.uniqueName}.pdf`);
+    }
+
+    /**
+     *  To convert base64 data to contetType format in chunks
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts:263]: Avoid incorrect spelling

File: apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts [hunk 2]
```diff
@@ -234,4 +246,92 @@ export class ViewTransactionsComponent implements OnInit, OnChanges, OnDestroy {
...
     }
+
+    /**
+    * download file as pdf
+    *
+    * @param data
+    * @param invoiceUniqueName
+    * @memberof ViewTransactionsComponent
+    */
+    public downloadFile() {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts:257]: return type

File: apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts [hunk 3]
```diff
@@ -234,4 +246,92 @@ export class ViewTransactionsComponent implements OnInit, OnChanges, OnDestroy {
...
+    /**
+     *  To convert base64 data to contentType format in chunks
+     *
+     * @param {any} b64Data base64 data string
+     * @param {string} contentType type to covert file
+     * @param {number} sliceSize chunk size
+     * @returns
+     * @memberof ViewTransactionsComponent
+     */
+    public base64ToBlob(b64Data: any, contentType: string, sliceSize: number) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts:271]: return type

File: apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts [hunk 4]
```diff
@@ -234,4 +246,92 @@ export class ViewTransactionsComponent implements OnInit, OnChanges, OnDestroy {
...
+        return new Blob(byteArrays, { type: contentType });
+    }
+
+    /**
+     * To send mail or download voucher
+     *
+     * @param {{ action: string, emails: string[], numbers: string[], typeOfInvoice: string[] }} userResponse API call object body
+     * @memberof ViewTransactionsComponent
+     */
+    public onDownloadOrSendMailEvent(userResponse: { action: string, emails: string[], numbers: string[], typeOfInvoice: string[] }) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts:298]: return type

File: apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts [hunk 5]
```diff
@@ -234,4 +246,92 @@ export class ViewTransactionsComponent implements OnInit, OnChanges, OnDestroy {
...
+        }
+    }
+
+    /**
+     * To download invoice service call
+     *
+     * @param {*} invoiceCopy
+     * @memberof ViewTransactionsComponent
+     */
+    public ondownloadInvoiceEvent(invoiceCopy) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.ts:319]: return type

**Pull request merged**